### PR TITLE
PLAT-1035 Open AutoCompletePopup on click into input

### PR DESCRIPTION
**Pull request by daniel-klose**
https://user-images.githubusercontent.com/90852097/186885423-496b8ad7-d94f-40f2-b1c1-1ecc233d2943.mp4

### CODE REVIEW

File: platform-dom/src/test/java/com/softicar/platform/dom/elements/input/auto/DomAutoCompletePopupTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+import com.softicar.platform.dom.elements.testing.engine.IDomTestExecutionEngine;
+import com.softicar.platform.dom.elements.testing.engine.IDomTestExecutionEngineMethods;
+import com.softicar.platform.dom.elements.testing.engine.document.DomDocumentTestExecutionEngine;
+import com.softicar.platform.dom.elements.testing.node.tester.DomNodeTester;
+import com.softicar.platform.dom.event.DomEventType;
+import com.softicar.platform.dom.input.DomTextInput;
+import org.junit.Rule;
+import org.junit.Test;
+
+public class DomAutoCompletePopupTest extends AbstractDomAutoCompleteDefaultInputEngineTest implements IDomTestExecutionEngineMethods {
```
Comment (daniel-klose) [inline platform-dom/src/test/java/com/softicar/platform/dom/elements/input/auto/DomAutoCompletePopupTest.java:13]: Hm, do I even need this test?
Wrote this before knowing all the Ajax tests

File: platform-ajax/src/test/java/com/softicar/platform/ajax/input/auto/complete/entity/AjaxAutoCompleteEntityInputFocusedPopupUnopenedTest.java [hunk 2]
```diff
@@ -23,7 +23,7 @@ public void testValidInputWithEnter() {
 			.execute();
 
 		input//
-			.focusByClick()
+			.clickInputField()
```
Comment (oliver-richers) [inline platform-ajax/src/test/java/com/softicar/platform/ajax/input/auto/complete/entity/AjaxAutoCompleteEntityInputFocusedPopupUnopenedTest.java:26]: shouldn't this be clickInputFieldAndClosePopup?

File: platform-ajax/src/test/java/com/softicar/platform/ajax/input/auto/complete/entity/AjaxAutoCompleteEntityInputFocusedPopupUnopenedTest.java [hunk 3]
```diff
@@ -90,7 +91,7 @@ public void testIllegalInputWithTab() {
...
```
Comment (oliver-richers) [inline platform-ajax/src/test/java/com/softicar/platform/ajax/input/auto/complete/entity/AjaxAutoCompleteEntityInputFocusedPopupUnopenedTest.java:94]: shouldn't this be clickInputFieldAndClosePopup?

File: platform-dom/src/test/java/com/softicar/platform/dom/elements/input/auto/DomAutoCompletePopupTest.java (hunk 1 above)
Comment (oliver-richers) [reply platform-dom/src/test/java/com/softicar/platform/dom/elements/input/auto/DomAutoCompletePopupTest.java:13]: not sure, we should ask Alex

**Review by oliver-richers — CHANGES_REQUESTED** [review]

File: platform-ajax/src/test/java/com/softicar/platform/ajax/input/auto/complete/AjaxAutoCompleteChangeEventTest.java [hunk 4]
```diff
@@ -220,6 +220,11 @@ public void testBackdropIfAlreadySubmitted() {
 		click(inputField);
 		waitForServer();
 
+		assertTrue(isAutoCompleteBackdropDisplayed());
+
+		click(inputField);
+		waitForServer();
+
 		assertFalse(isAutoCompleteBackdropDisplayed());
```
Comment (oliver-richers) [inline platform-ajax/src/test/java/com/softicar/platform/ajax/input/auto/complete/AjaxAutoCompleteChangeEventTest.java:228]: as discussed, it should not close when clicked again

**Review by alexander-schmidt — APPROVED** [review]

**Review by oliver-richers — APPROVED** [review]

**Pull request merged**